test/TestFilePathEvaluation.py

```diff
@@ -0,0 +1,120 @@
+"""Testing file path evaluation when using import_tasks / include_tasks."""
+import textwrap
+
+import pytest
+
+from ansiblelint.runner import Runner
+
+LAYOUT_IMPORTS = {
+    'main.yml': textwrap.dedent(
+        """\
+        ---
+        - hosts: target
+          gather_facts: false
+          tasks:
+            - name: from main import task 1
+              import_tasks: tasks/task_1.yml
+        """
+    ),
+    'tasks/task_1.yml': textwrap.dedent(
+        """\
+        ---
+        - name: from task 1 import task 2
+          import_tasks: tasks/task_2.yml
+        """
+    ),
+    'tasks/task_2.yml': textwrap.dedent(
+        """\
+        ---
+        - name: from task 2 import subtask 1
+          import_tasks: tasks/subtasks/subtask_1.yml
+        """
+    ),
+    'tasks/subtasks/subtask_1.yml': textwrap.dedent(
+        """\
+        ---
+        - name: from subtask 1 import subtask 2
+          import_tasks: tasks/subtasks/subtask_2.yml
+        """
+    ),
+    'tasks/subtasks/subtask_2.yml': textwrap.dedent(
+        """\
+        ---
+        - name: from subtask 2 do something
+          debug:
+            msg: |
+              Something...
+        """
+    ),
+}
+
+LAYOUT_INCLUDES = {
+    'main.yml': textwrap.dedent(
+        """\
+        ---
+        - hosts: target
+          gather_facts: false
+          tasks:
+            - name: from main import task 1
+              include_tasks: tasks/task_1.yml
+        """
+    ),
+    'tasks/task_1.yml': textwrap.dedent(
+        """\
+        ---
+        - name: from task 1 import task 2
+          include_tasks: tasks/task_2.yml
+        """
+    ),
+    'tasks/task_2.yml': textwrap.dedent(
+        """\
+        ---
+        - name: from task 2 import subtask 1
+          include_tasks: tasks/subtasks/subtask_1.yml
+        """
+    ),
+    'tasks/subtasks/subtask_1.yml': textwrap.dedent(
+        """\
+        ---
+        - name: from subtask 1 import subtask 2
+          include_tasks: tasks/subtasks/subtask_2.yml
+        """
+    ),
+    'tasks/subtasks/subtask_2.yml': textwrap.dedent(
+        """\
+        ---
+        - name: from subtask 2 do something
+          debug:
+            msg: |
+              Something...
+        """
+    ),
+}
+
+
+@pytest.mark.parametrize(
+    'ansible_project_layout',
+    (
+        pytest.param(LAYOUT_IMPORTS, id='using only import_tasks'),
```

Comment:
Usually I prefer test ids that do not have spaces or special chars, so I can use them with `-k` without trouble but I will likely fix it next time we need to touch the test, hopefully soon for removal of xfail.

Reply:
FWIW spaced ids work with `-k` just fine. But I prefer concrete specifiers like `tox -- 'test/TestFilePathEvaluation.py::test_file_path_evaluation[using only include_tasks]'`